File: Python/valid_anagram.py
def validAnagram(s: str, t: str) -> bool:
    if(len(s) != len(t)):
        return False

    countS, countT = {}, {}

    for i in range(len(s)):
        countS[s[i]] = 1 + countS.get(s[i], 0)
        countT[t[i]] = 1 + countT.get(t[i], 0)
    return countT == countS


s = "anagram"
t = "naagram"

s2 = "rat"
t2 = "cat"

print(validAnagram(s, t))
print(validAnagram(s2, t2))

File: C++/valid_anagram.cc
#include <bits/stdc++.h>

bool valid_anagram(std::string s, std::string t)
{
    if (s.size() != t.size())
        return false;

    std::unordered_map<char, int> s_map;
    std::unordered_map<char, int> t_map;

    for (int i = 0; i < s.size(); i++)
    {
        s_map[s[i]]++;
        t_map[t[i]]++;
    }

    for (int i = 0; i < s_map.size(); i++)
    {
        if (s_map[i] != t_map[i])
            return false;
    }
    return true;
}

int main()
{
    std::string s{"anagram"}, t{"nagaram"};
    std::string s2{"car"}, t2{"rat"};
    std::cout << valid_anagram(s, t) << std::endl;
    std::cout << valid_anagram(s2, t2) << std::endl;
}